tests/test_sign_tx_cmd.py

```diff
@@ -0,0 +1,272 @@
+                NavInsID.USE_CASE_STATUS_DISMISS
+            ],
+            [
+                NavInsID.USE_CASE_REVIEW_TAP,
+                NavInsID.USE_CASE_REVIEW_REJECT,
+                NavInsID.USE_CASE_CHOICE_CONFIRM,
+                NavInsID.USE_CASE_STATUS_DISMISS
+            ],
+            [
+                NavInsID.USE_CASE_REVIEW_TAP,
+                NavInsID.USE_CASE_REVIEW_TAP,
+                NavInsID.USE_CASE_REVIEW_REJECT,
+                NavInsID.USE_CASE_CHOICE_CONFIRM,
+                NavInsID.USE_CASE_STATUS_DISMISS
+            ],
+            [
+                NavInsID.USE_CASE_REVIEW_REJECT,
+                NavInsID.USE_CASE_CHOICE_REJECT,
+                NavInsID.USE_CASE_REVIEW_REJECT,
+                NavInsID.USE_CASE_CHOICE_CONFIRM,
+                NavInsID.USE_CASE_STATUS_DISMISS
+            ]
+        ]
+
+        for i, instructions in enumerate(instructions_list):
+            # Send the sign device instruction.
+            # As it requires on-screen validation, the function is asynchronous.
+            # It will yield the result when the navigation is done
+            with client.sign_tx(path=path, transaction=transaction):
+                # Validate the on-screen request by performing the navigation appropriate for this device
+                navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH, test_name + f"/part{i}", instructions)
+
+            # The device as yielded the result, parse it and ensure that the signature is correct
+            response = client.get_async_response()
+
+            # Assert that we have received a refusal
+            assert response.status == Errors.SW_TRANSACTION_CANCELLED
+            assert len(response.data) == 0
+
+
+# In this test we send to the device a transaction (including multiple clauses and data)
+# to sign and validate it on screen
+# The transaction is short and will be sent in one chunk
+# We will ensure that the displayed information is correct by using screenshots comparison
+def test_sign_tx_short_tx_data_and_multiple_clauses(firmware, backend, navigator, test_name):
+    if not isinstance(backend, SpeculosBackend):
+        input("Please confirm that multi-clauses and contract data are enabled in app settings?")
+
+    # Use the app interface instead of raw interface
+    client = VechainClient(backend)
+
+    if firmware.device.startswith("nano"):
+        # first activate multiple cause and data settings
+        instructions = [
+            NavInsID.RIGHT_CLICK,
+            NavInsID.BOTH_CLICK,
+            NavInsID.BOTH_CLICK,
+            NavInsID.RIGHT_CLICK,
+            NavInsID.BOTH_CLICK,
+            NavInsID.RIGHT_CLICK,
+            NavInsID.BOTH_CLICK,
+            NavInsID.RIGHT_CLICK,
+            NavInsID.BOTH_CLICK,
+            NavInsID.RIGHT_CLICK,
+            NavInsID.BOTH_CLICK
+        ]
+
+        navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH, test_name, instructions,
```

Comment:
Nano device tests stopped here

Reply:
@libotony sorry not sure to understand your issue here., could you please elaborate ? On my side all tests are running well for all devices.

Reply:
https://github.com/LedgerHQ/app-vechain/tree/TJ/add_tests/tests/snapshots/nanos/test_sign_tx_short_tx_data_and_multiple_clauses
It stops after changing the settings, I was expecting tx action to be performed.

Reply:
ok I

Reply:
Ok I understand now. In this test, the snapshots are only created when the settings are changed and then I use the ragger function based on OCR (but without snapshots) to check if I have the expected screens.


        # Send the sign device instruction.
        # As it requires on-screen validation, the function is asynchronous.
        # It will yield the result when the navigation is done

        with client.sign_tx(path=path, transaction=transaction_multi_clauses_and_data[0]):

            navigator.navigate_until_text(NavInsID.RIGHT_CLICK,
                                            [NavInsID.BOTH_CLICK],
                                            "Data present",
                                            screen_change_before_first_instruction=False)
            navigator.navigate_until_text(NavInsID.RIGHT_CLICK,
                                            [NavInsID.BOTH_CLICK],
                                            "Multiple Clauses",
                                            screen_change_before_first_instruction=False)

            navigator.navigate_until_text(NavInsID.RIGHT_CLICK,
                                            [NavInsID.BOTH_CLICK],
                                            "Accept",
                                            screen_change_before_first_instruction=False)
`

Reply:
I see, I missed the `_compare` part, thanks